src/ppx/scene/scene_resource_manager.cpp

```diff
@@ -98,5 +98,41 @@ void ResourceManager::DestroyAll()
     mMeshes.clear();
 }
 
+std::vector<const scene::Sampler*> ResourceManager::GetSamplers() const
+{
+    std::vector<const scene::Sampler*> objectPtrs;
+    for (auto& it : mSamplers) {
+        objectPtrs.push_back(it.second.get());
+    }
+    return objectPtrs;
+}
+
+std::vector<const scene::Image*> ResourceManager::GetImages() const
+{
+    std::vector<const scene::Image*> objectPtrs;
+    for (auto& it : mImages) {
+        objectPtrs.push_back(it.second.get());
```

Comment:
This is traversing unordered maps.  Does it matter that separate invocations may change the order in which these vectors are built up?

Reply:
It shouldn't matter. The  calling code uses the vector as the basis for indexing. An index map is built from that which gets fed into the shaders.